File: scripts/add_emojis.js
const fs = require('fs');
const path = require('path');

const docsDir = '/home/than/Personal Project/odoo/clone gitbook to fumadoc/fumadocs-clone/content/docs';

const emojiMap = [
    { keywords: ['Al-Quran', 'Al Qur\'an', 'Surah', 'Ayat'], emoji: '📖' },
    { keywords: ['Tahfidz', 'Tahsin'], emoji: '📖' },
    { keywords: ['Absensi', 'Kehadiran'], emoji: '✅' },
    { keywords: ['Penilaian', 'Nilai', 'Rapor'], emoji: '📊' },
    { keywords: ['Login', 'Akun', 'Hak Akses', 'Pemblokiran', 'Pengaktifan'], emoji: '🔐' },
    { keywords: ['Santri', 'Siswa'], emoji: '👨‍🎓' },
    { keywords: ['Guru', 'Wali Kelas', 'Musyrif'], emoji: '👨‍🏫' },
    { keywords: ['Hadits', 'Buku'], emoji: '📖' },
    { keywords: ['Kesehatan'], emoji: '🏥' },
    { keywords: ['Pelanggaran', 'Tindakan Disipliner'], emoji: '⚠️' },
    { keywords: ['Keuangan', 'Biaya', 'Tagihan', 'Saldo', 'VA', 'Donasi', 'Mata Uang', 'Pajak', 'Akuntansi', 'Order Pembelian', 'Tagihan Pemasok'], emoji: '💰' },
    { keywords: ['Pengaturan', 'Konfigurasi', 'Setup', 'Alur', 'Role'], emoji: '⚙️' },
    { keywords: ['Laporan'], emoji: '📋' },
    { keywords: ['Dashboard', 'UI', 'Tampilan', 'Tampilan Sistem'], emoji: '🖥️' },
    { keywords: ['Jadwal', 'Jam Pelajaran'], emoji: '📅' },
    { keywords: ['Pendaftaran', 'PSB', 'Kartu'], emoji: '📝' },
    { keywords: ['Gudang', 'Penyimpanan', 'Lokasi'], emoji: '📦' },
    { keywords: ['Point of Sale', 'POS', 'Kasir', 'Penjualan'], emoji: '🛒' },
    { keywords: ['Halaqah', 'Asrama', 'Kamar'], emoji: '🏠' },
    { keywords: ['Pembelian', 'Pemasok'], emoji: '🤝' },
    { keywords: ['Struktur Organisasi'], emoji: '🏢' },
    { keywords: ['Jenjang', 'Jurusan', 'Kelas', 'Mata Pelajaran', 'Kurikulum', 'RPP'], emoji: '🏫' },
    { keywords: ['Pengumuman', 'Informasi'], emoji: '📢' },
];

function getEmoji(content, title, fileName) {
    const searchTitle = (title + ' ' + fileName).toLowerCase();
    const searchContent = content.toLowerCase();

    // High priority: Title match
    for (const entry of emojiMap) {
        for (const keyword of entry.keywords) {
            if (searchTitle.includes(keyword.toLowerCase())) {
                return entry.emoji;
            }
        }
    }

    // Low priority: Content match
    for (const entry of emojiMap) {
        for (const keyword of entry.keywords) {
            if (searchContent.includes(keyword.toLowerCase())) {
                return entry.emoji;
            }
        }
    }

    return '📄';
}

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');

    // Extract title from frontmatter
    const frontmatterMatch = content.match(/^---\s*([\s\S]*?)\s*---/);
    let title = '';
    if (frontmatterMatch) {
        const frontmatter = frontmatterMatch[1];
        const titleMatch = frontmatter.match(/^title:\s*(.*)$/m);
        if (titleMatch) {
            title = titleMatch[1].trim();
            // Remove existing emoji if present
            title = title.replace(/^[\u{1F300}-\u{1F6FF}\u{2600}-\u{26FF}\u{2700}-\u{27BF}\u{1F900}-\u{1F9FF}\u{1F600}-\u{1F64F}\u{1F680}-\u{1F6CF}\u{1F300}-\u{1F5FF}\u{1F900}-\u{1F9FF}\u{1F1E0}-\u{1F1FF}\u{1F170}-\u{1F251}]\s*/u, '');
        }
    }

    const fileName = path.basename(filePath, path.extname(filePath));
    if (!title) {
        title = fileName.split('-').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(' ');
    }

    const emoji = getEmoji(content, title, fileName);
    const h1Pattern = /^#\s+(.*)$/m;
    const h1Match = content.match(h1Pattern);

    const newH1 = `# ${emoji} ${title}`;

    if (h1Match) {
        // Replace existing H1
        content = content.replace(h1Pattern, newH1);
    } else {
        // Insert after frontmatter
        if (frontmatterMatch) {
            content = content.replace(/^---\s*[\s\S]*?\s*---/, (match) => `${match}\n\n${newH1}`);
        } else {
            content = `${newH1}\n\n${content}`;
        }
    }

    // Clean up double newlines
    content = content.replace(/\n{3,}/g, '\n\n');

    fs.writeFileSync(filePath, content);
    console.log(`Processed: ${filePath} -> ${emoji}`);
}

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (file.endsWith('.md') || file.endsWith('.mdx')) {
            processFile(fullPath);
        }
    }
}

walkDir(docsDir);
console.log('Done!');

File: scripts/add_emojis.py
import os
import re

def get_emoji(path, title):
    path = path.lower()
    title = title.lower()
    
    # Priority keywords in title
    if 'absensi' in title: return '📝'
    if 'penilaian' in title: return '🎓'
    if 'kesehatan' in title: return '🏥'
    if 'pelanggaran' in title: return '⚠️'
    if 'prestasi' in title: return '🏆'
    if 'izin' in title: return '🎫'
    if 'saldo' in title or 'keuangan' in title: return '💰'
    if 'tagihan' in title or 'biaya' in title: return '🧾'
    if any(k in title for k in ['tahfidz', 'tahsin', 'quran']): return '📖'
    if 'pembelian' in title or 'order' in title: return '🛒'
    if 'pos' in title or 'kasir' in title: return '🏪'
    if 'login' in title or 'masuk' in title: return '🔑'
    
    # Path based mapping
    if 'modul-guru' in path: return '👨‍🏫'
    if 'modul-keamanan' in path: return '👮'
    if 'modul-kesantrian' in path: return '🏫'
    if 'modul-keuangan' in path: return '💰'
    if 'modul-musyrif' in path: return '👳'
    if 'modul-orang-tua' in path: return '👨‍👩‍👧'
    if 'modul-pembelian' in path: return '🛒'
    if 'modul-pos' in path: return '🏪'
    if 'modul-psb' in path: return '📝'
    if 'dashboard' in path: return '📊'
    if 'data-riwayat' in path: return '📜'
    if 'laporan-laporan' in path: return '📈'
    if 'master-data' in path: return '🗂️'
    if 'panduan-login' in path: return '🔑'
    if 'pengenalan-sistem' in path: return '💡'
    if 'setup-and-konfigurasi' in path: return '⚙️'
    
    return '📄'

def update_mdx_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Extract frontmatter
    match = re.search(r'^---\s*\n(.*?)\n---\s*\n', content, re.DOTALL)
    if not match:
        print(f"No frontmatter found in {filepath}")
        return

    frontmatter_content = match.group(1)
    body = content[match.end():]
    
    # Extract title
    title_match = re.search(r'^title:\s*(.*)$', frontmatter_content, re.MULTILINE)
    title = title_match.group(1).strip() if title_match else ""
    # Remove quotes if present
    title = title.strip('"').strip("'")
    
    # Determine emoji
    emoji = get_emoji(filepath, title)
    
    # Check if icon already exists
    if re.search(r'^icon:', frontmatter_content, re.MULTILINE):
        # Update existing icon
        updated_frontmatter = re.sub(r'^icon:.*$', f'icon: {emoji}', frontmatter_content, flags=re.MULTILINE)
    else:
        # Add icon after title or at the end of frontmatter
        if title_match:
            updated_frontmatter = frontmatter_content[:title_match.end()] + f'\nicon: {emoji}' + frontmatter_content[title_match.end():]
        else:
            updated_frontmatter = frontmatter_content + f'\nicon: {emoji}'
            
    # Combine back
    new_content = f"---\n{updated_frontmatter.strip()}\n---\n{body}"
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print(f"Updated {filepath} with icon: {emoji}")

def main():
    docs_dir = 'content/docs'
    for root, dirs, files in os.walk(docs_dir):
        for file in files:
            if file.endswith('.mdx'):
                update_mdx_file(os.path.join(root, file))

if __name__ == "__main__":
    main()
